tests/Functional/Excecution/ExecutionTest.php

```diff
@@ -0,0 +1,310 @@
+                                            'value' => 'name'
+                                        ]),
+                                        'type' => GraphQLString(),
+                                        'defaultValue' => 'Han Solo'
+                                    ])
+                                ]
+                            ])
+                        ]
+                    ]),
+                    'operation' => 'query',
+                    'directives' => [],
+                    'variableDefinitions' => []
+                ])
+            ],
+        ]);
+
+        $rootValue      = [];
+        $contextValue   = '';
+        $variableValues = [];
+        $operationName  = 'query';
+        $fieldResolver  = null;
+
+        /** @var ExecutionResult $executionResult */
+        $executionResult = Execution::execute(
+            $schema,
+            $documentNode,
+            $rootValue,
+            $contextValue,
+            $variableValues,
+            $operationName,
+            $fieldResolver
+        );
+
+        $expected = new ExecutionResult(['greeting' => 'Hello Han Solo'], []);
+
+        $this->assertEquals($expected, $executionResult);
+    }
+
+    /**
+     * @throws \TypeError
+     */
+    public function testExecuteQueryWithMultipleFields()
+    {
+        $schema = new Schema([
+            'query' =>
+                new ObjectType([
+                    'name'   => 'Human',
+                    'fields' => [
+                        'id' =>  [
+                            'type'    => GraphQLInt(),
+                            'resolve' => function () {
+                                return 1000;
+                            }
+                        ],
+                        'type' => [
+                            'type'    => GraphQLString(),
+                            'resolve' => function () {
+                                return 'Human';
+                            }
+                        ],
+                        'friends' => [
+                            'type'    => GraphQLList(GraphQLString()),
+                            'resolve' => function () {
+                                return ['1002', '1003', '2000', '2001'];
+                            }
+                        ],
+                        'appearsIn' => [
+                            'type' => GraphQLList(GraphQLInt()),
+                            'resolve' => function () {
+                                return [4, 5, 6];
+                            }
+                        ],
+                        'homePlanet' => [
+                            'type' => GraphQLString(),
+                            'resolve' => function () {
+                                return 'Tatooine';
+                            }
+                        ],
+                    ]
+                ])
+        ]);
+
+        $documentNode = new DocumentNode([
```

Comment:
I'm working on a `toArray` functionality for the node tree. This will help testing node tree construction when parsing a GQL string.